packages/frontend/src/lib/pnl.ts

```diff
@@ -0,0 +1,95 @@
+import BigNumber from 'bignumber.js'
+
+import { TransactionType } from '@constants/enums'
+import { PositionType } from '../types'
+
+interface Deposit {
+  squeethAmount: BigNumber
+  ethAmount: BigNumber
+  usdValue: BigNumber
+  timestamp: string
+  transactionType: TransactionType
+  txId: string
+  ethPriceAtDeposit: BigNumber
+}
+
+interface LongDeposit extends Deposit {
+  sellQuote: BigNumber
+}
+
+interface GetCurrentLongParams {
+  positionType: string
+  squeethAmount: BigNumber
+  transactions: any[]
+  getSellQuote: (input: BigNumber) => Promise<Record<string, any>>
+}
+
+export async function getCurrentLong({
+  getSellQuote,
+  positionType,
+  squeethAmount,
+  transactions,
+}: GetCurrentLongParams) {
+  if (positionType === PositionType.SHORT) return []
+  let totalLongSqth = new BigNumber(0)
+
+  const result: LongDeposit[] = []
+  let sellQuote
+  for (let index = 0; index < transactions.length; index++) {
+    if (totalLongSqth.gte(squeethAmount)) break
+    if (totalLongSqth.isLessThan(squeethAmount) && transactions[index].transactionType === TransactionType.BUY) {
+      totalLongSqth = totalLongSqth.plus(transactions[index].squeethAmount)
+      sellQuote = await getSellQuote(transactions[index].squeethAmount)
+      result.push({ ...transactions[index], sellQuote: sellQuote.amountOut })
+    } else if (
+      totalLongSqth.isLessThan(squeethAmount) &&
+      transactions[index].transactionType === TransactionType.SELL
+    ) {
+      totalLongSqth = totalLongSqth.minus(transactions[index].squeethAmount)
+    }
+  }
+
+  return result
+}
+
+interface ShortDeposit extends Deposit {
+  buyQuote: BigNumber
+}
+
+interface GetCurrentShortParams {
+  positionType: string
+  squeethAmount: BigNumber
+  transactions: any[]
+  getBuyQuote: (input: BigNumber) => Promise<Record<string, any>>
+}
+
+export async function getCurrentShortDeposits({
+  getBuyQuote,
+  positionType,
+  squeethAmount,
+  transactions,
+}: GetCurrentShortParams) {
+  if (positionType === PositionType.LONG) return []
+  let totalShortSqth = new BigNumber(0)
+
+  const result: ShortDeposit[] = []
+  let buyQuote
+  for (let index = 0; index < transactions.length; index++) {
+    if (totalShortSqth.gte(squeethAmount)) break
+    if (
+      totalShortSqth.isLessThan(squeethAmount) &&
```

Comment:
User may mint and open short in two separate transactions, if we just check for MINT_SHORT and BURN_SHORT, then we will miss calculate some transaction
So does burn and buy back

Reply:
We can use the short helper does minting + selling in 1 tx vs 2 different tx for the mint on fe and sell on uniswap. There should be different addresses involved as well (Controller + uniswap pool directly from user address vs ShortHelper calling controller + uniswap from itself) -- quoted from Andrew lol

Reply:
MINT_SHORT is one tx, so does BURN_SHORT

Reply:
@alexisgauba wanna make sure that I understand positions correctly

Reply:
Also have to take adding & removing liquidity into account

Reply:
Yes people can go short and close short through the short helper but many people have also been doing it manually by minting and then selling on uniswap or buying back on uniswap and then burning

Reply:
Don't think adding / removing liquidity is needed here? We're just trying to show unrealized PnL for partially closed positions, so if someone closes their long partially or short partially

Reply:
But if we dont take them into account, will the squeeth amount be accurate?